main.tf

```diff
@@ -89,9 +89,14 @@ resource "google_cloud_run_v2_job" "job" {
   }
 }
 
+module "delivery-sa-name" {
+  source = "./short-unique-name"
+  name   = "${var.name}"
+}
```

Comment:
I'd just forgo the module entirely and follow the pattern from mono:
```hcl
// Compute a suffix that satisfies the regex:
// ^[a-z](?:[-a-z0-9]{4,28}[a-z0-9])$
resource "random_string" "bar" {
  length  = 30 - length(local.sa_prefix)
  special = false
  upper   = false
}

// Create a dedicated GSA for the foo bar service.
resource "google_service_account" "bar" {
  project = var.project_id

  account_id   = "${local.sa_prefix}${random_string.bar.result}"
  display_name = "Foo Bar"
  description  = "Dedicated service account for the Foo bar service"
}
```

Reply:
ref: https://github.com/chainguard-dev/mono/blob/main/docs/iac-docs/service-accounts.md

Reply:
I want to trim it too in case the provided prefix already was too long